tests/csm/rest/test_capacity_quota.py

```diff
@@ -1927,3 +1927,119 @@ def test_40605(self):
+        enabled = test_cfg["enabled"]
+        max_size = test_cfg["max_size"]
+        max_objects = test_cfg["max_objects"]
+        self.log.info("Step 1: Set max_size and max_objects quota for user")
+        payload = self.csm_obj.iam_user_quota_payload(enabled, max_size, max_objects,
+                                                      check_on_raw=True)
+        result, resp = self.csm_obj.verify_get_set_user_quota(self.user_id, payload,
+                                                              verify_response=True)
+        assert result, "Verification for get set user failed."
+        self.log.info("Response : %s", resp)
+        random_size = self.csm_obj.random_gen.randrange(1, max_size)
+        remaining_size = max_size - random_size
+        size_list = [random_size, remaining_size]
+        obj_list = []
+        self.log.info("Step 2: Try uploading 1 object(obj-1) of less than max_size"
+                      "and remaining size")
+        for size in size_list:
+            self.log.info("Uploading %s size object", size)
+            resp = s3_misc.create_put_objects(self.obj_name, self.bucket,
+                                          self.akey, self.skey, object_size=int(size/1024),
+                                          block_size="1K")
+            assert resp, "Put object Failed"
+            obj_list.append(self.obj_name)
+        self.log.info("Step 3: Repeat above step")
+        try:
+            resp = s3_misc.create_put_objects(self.obj_name, self.bucket,
+                                          self.akey, self.skey,
+                                 object_size=int(remaining_size/1024), block_size="1K")
+        except ClientError as error:
+            self.log.info("Expected exception received %s", error)
+            assert error.response['Error']['Code'] == "QuotaExceeded", \
+                                      "Overwriting object passed"
+        obj_name = f'{self.obj_name_prefix}{time.perf_counter_ns()}'
+        self.log.info("Step 4: Try uploading one more object with different name and size")
+        resp = s3_misc.create_put_objects(obj_name, self.bucket,
+                                          self.akey, self.skey,
+                                   object_size=int(remaining_size/1024), block_size="1K")
+        assert resp, "Put Object Failed"
+        obj_list.append(obj_name)
+        self.log.info("Step 5: Try overwriting above object")
+        try:
+            resp = s3_misc.create_put_objects(obj_name, self.bucket,
+                                          self.akey, self.skey,
+                                 object_size=int(remaining_size/1024), block_size="1K")
+        except ClientError as error:
+            self.log.info("Expected exception received %s", error)
+            assert error.response['Error']['Code'] == "QuotaExceeded", \
+                                      "Second overwrite passed"
+        self.log.info("Step 6: Try uploading one more object")
```

Comment:
Can we add loop for different sizes which are expected to fail?

Reply:
any more objects upload of any size would fail after this step.

Reply:
There are duplicate try except block which can be put in loop.